monitor.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ebiten
+
+import (
+	"image"
+
+	"github.com/hajimehoshi/ebiten/v2/internal/ui"
+)
+
+// MonitorID represents a monitor's identifier.
+type MonitorID int
+
+// Monitor represents a monitor available to the system.
+type Monitor struct {
+	id            MonitorID
+	name          string
+	x, y          int
+	width, height int
+}
+
+// Bounds returns the position and size of the monitor.
+func (m *Monitor) Bounds() image.Rectangle {
+	return image.Rect(m.x, m.y, m.x+m.width, m.y+m.height)
+}
+
+// ID returns the monitor's ID. 0 is always the primary monitor.
+func (m *Monitor) ID() MonitorID {
```

Comment:
This needs some more thought, as `WindowMonitor()` should have an ID associated with the returned Monitor struct, otherwise there is no point to such a function. Additionally, this allows developers to properly indicate which monitor is active, according to ebitengine's view. At the moment the codepath for `WindowMonitor()` doesn't properly populate this ID at the moment, but it should.

Reply:
Alternatively, the Monitor struct could have an `Active() bool` func that just returns if it is the active monitor or not.

Reply:
I'm afraid I was missing the fact that ebiten package treats `Monitor` as a value, not a pointer to a value.

Then, `ui.WindowMonitor` and `ebiten.WindowMonitor` should return a pointer to a struct. Also,

```
WindowMonitor() Monitor
SetWindowMonitor(monitor Monitor)
AppendMonitors(monitors []Monitor) []Monitor
```

should be

```
WindowMonitor() *Monitor
SetWindowMonitor(monitor *Monitor)
AppendMonitors(monitors []*Monitor) []*Monitor
```

and the definition of ebiten.Monitor should be

```
type Monitor ui.Monitor

// Method definitions are still requried for docmentations.
func (m *Monitor) Bounds() image.Rectangle ...
```

`uiMonitorToMonitor` is no longer needed since this can be converted easily:

```
var x *ui.Monitor
var y *Monitor
y = (*Monitor)(x)
```

Reply:
Hmm, okay, so checking for "active" monitor can now be resolved via:
```golang
	activeMonitor := ebiten.WindowMonitor()
	for i, m := range ebiten.AppendMonitors(nil) {
		if m == activeMonitor {
			text.Draw(screen, fmt.Sprintf("active: %s (%d)", m.Name(), i), mplusNormalFont, x, y, color.White)
		}
	}
```

Reply:
Yes!